task6/script.js

```diff
@@ -0,0 +1,293 @@
+            createdAt: new Date('2019-03-17T23:00:00'),
+            author: 'Иванов Иван'
+        },
+
+        {
+            id: '19',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2019-03-17T23:00:00'),
+            author: 'Николай'
+        },
+
+        {
+            id: '20',
+            description: 'Более 76 тыс. человек во всем мире уже излечились от заболевания, спровоцированного новым коронавирусом, тогда как количество смертей превысило 6,4 тыс.',
+            createdAt: new Date('2021-03-17T23:00:00'),
+            author: 'Николай'
+        },
+
+    ];
+
+    
+    function getPost(id) {
+        for (let i = 0; i < posts.length ; i++) {
+            if (posts[i].id === id) {
+                return posts[i];
+            }
+        }
+    }
+    
+    function validatePost(post){
+        if(typeof post.id !== 'string' || post.id === ''){
+            return false;
+        }
+        if(typeof post.description !== 'string' || post.description === '' || post.description.length > 200)
+            return false;
+        if(!(post.createdAt instanceof Date)){
+            return false;
+        }
+        if(typeof post.author !== 'string' || post.author === ''){
+            return false;
+        }
+        return true;
+    }
+    
+    function addPost(post) {
+        if (validatePost(post)) {            
+            posts.push(post);
+            return true;
+        }
+        return false;
+    }
+    
+    function removePost(id){
+        for(let i = 0; i < posts.length; i++) {
+            if(posts[i].id === id){
+                posts.splice(i,1);
+                return true;
+            }
+        }
+        return false;
+    }
+
+    
+    function editPost(id, post){
+        let editPost = getPost(id);
+        if(post.description !== '' && typeof post.description == 'string'){
+            editPost.description = post.description;
+        if(validatePost(editPost)){
+            for( let i = 0; i < posts.length; i++){
+                if(posts[i].id === id){
+                    posts[i] = editPost;
+                    return true;
+                }
+              }
+           }
+        }        
+        return false;
+    }
+    
+
+    function sortByDate(posts) {
+        posts.sort((post1,post2) => post1.createdAt - post2.createdAt);
+    }
+    
+    
+    function getPosts(skip = 0, top = 10, filterConfig) {
+        if(typeof skip !== 'number' || typeof top !== 'number'){
+            return 'некорректно заданы параметры';
+        }
+        let arr = [];
+        let searchArr = posts;
+        if (typeof filterConfig === 'undefined') {
+            arr = searchArr.slice(skip,skip+top);
+            sortByDate(arr);
```

Comment:
It should be firstly sorted and then sliced. Imagine case, when you have posts created at times `1, 3, 2`, if you'll take 2 top posts, you'll receive `1,3` instead of `1, 2`